lib/charted.js

```diff
@@ -10,15 +10,16 @@ module.exports = function(app, charted_root) {
 
   app.use(charted_root + 'get', function (req, res, next) {
     var get_url = url.parse(req.url, true).query.url
-
     if (!get_url) {
       res.statusCode = 400
       res.end('Bad Request: no url provided')
-      next()
       return
     }
 
-    request(decodeURIComponent(get_url), function (err, resp, body) {
+    var get_url_full = url.parse(get_url).host ? get_url :
+        ('http://' + req.headers.host + '/' + get_url.replace(/^\//, ''))
```

Comment:
@valueof: This seems to be as far as I can get in determining the absolute URL when this endpoint gets hit with a relative path, e.g. 'example.csv'. All of the ways I've looked at for obtaining the absolute URL seem to fail on Connect...

	console.log(req.protocol)
	console.log(req.url)
	console.log(req.originalUrl)
	console.log(req.host)
	console.log(req.headers.host)
	console.log(req.params)

You can verify this by changing index.js to initialize a Connect app instead of Express app.